### PR TITLE
Blog on bad blocks

### PR DESCRIPTION
**Pull request by ldorau**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmem.github.io/89)

### CODE REVIEW

File: _posts/2018-11-19-bad-blocks.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+$ pmempool sync --bad-blocks /dev/dax1.0
+error: failed to synchronize: a part of the pool has uncorrectable errors in all replicas
+error: Invalid argument
+```
+
+Fixing bad blocks causes creating or reading special recovery files.
+When bad blocks are detected, special recovery files have to be created
+in order to fix them safely. A separate recovery file is created per each part
+of the pool. The recovery files are created in the same directory
+where the poolset file is located using the following name pattern:
```
Comment (pbalcer) [inline _posts/2018-11-19-bad-blocks.md]: as the poolset file, using the following name pattern

File: _posts/2018-11-19-bad-blocks.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+author: ldorau
+layout: post
+identifier: bad_blocks
+---
+
+### Introduction
+
+Storage devices can contain uncorrectable media errors often called
+"bad blocks". A bad block is a part of a storage media that is either
+inaccessible or unwriteable due to a permanent physical damage.
```
[the remark this answers is not in the document]
Comment (ldorau) [reply _posts/2018-11-19-bad-blocks.md]: Done

File: _posts/2018-11-19-bad-blocks.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+If a process tries to access (read or write) the corrupted block,
+it will be terminated by the SIGBUS signal.
```
[the remark this answers is not in the document]
Comment (ldorau) [reply _posts/2018-11-19-bad-blocks.md]: Done

File: _posts/2018-11-19-bad-blocks.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+### Handling bad blocks in PMDK libraries
+
+PMDK libraries can handle bad blocks if the `CHECK_BAD_BLOCKS`
+compat feature is turned on. Currently (PMDK v1.5) it is disabled by default
+because it requires super user privileges. It can be turned on using
+[pmempool-feature](https://github.com/ldorau/pmem.github.io/blob/blog/_posts/2018-11-19-bad-blocks.md#pmempool-feature).
```
[the remark this answers is not in the document]
Comment (ldorau) [reply _posts/2018-11-19-bad-blocks.md]: Done

File: _posts/2018-11-19-bad-blocks.md (hunk 1 above)
Comment (ldorau) [reply _posts/2018-11-19-bad-blocks.md]: Done

**Pull request merged**